server.js

```diff
@@ -6,22 +6,100 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavaliable" })
+  }
+});
+
+mongoose.set('toJSON', {
+  virtuals: true,
+  versionKey: false,
+  transform: (doc, converted) => {
+    delete converted._id;
+  }
+});
```

Comment:
I don't fully understand this code but assume it is for removing the _ from the id. Nice perseverance on getting that done! 😎